src/vellum/workflows/runner/runner.py

```diff
@@ -140,6 +145,19 @@ def __init__(
                 if issubclass(ei.inputs_class.__parent_class__, BaseNode)
             ]
             self._is_resuming = True
+        elif previous_execution_id:
+            self.previous_execution_id = str(previous_execution_id)
+            for resolver in self.workflow.resolvers:
+                try:
+                    load_state_result = resolver.load_state(self.previous_execution_id)
+                    if load_state_result.state is not None:
+                        self._initial_state = cast(StateType, load_state_result.state)
```

Comment:
[nit] we should almost never cast. In this case, we should ensure that `isinstance(load_state_result.state, StateType)`